### PR TITLE
[MGDOBR-966] - Remove hardcoded default namespace reference

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
https://issues.redhat.com/browse/MGDOBR-966

**Expectation**
For manage-tenant on-boarding, generated operator-bundle should be independent of namespaces. Namespace value will be injected while creating `addon`.

**Actual**
With the current implementation, namespace attribute always injected in to Deployment and ServiceAccount while compiling project using maven.

**Solution**
Remove hardcoded default namespace reference from pom.xml and only provide namespace value while compiling project using
`-Dquarkus.kubernetes.namespace=$NAMESPACE` argument.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: .github/workflows/Operator-CI.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
       - name: Build Shard operator image and resources
         uses: ./.github/actions/maven
         with:
-          maven-command: clean install --errors --quiet -Pminikube -Dquarkus.container-image.tag=$GITHUB_SHA -Dnamespace=$NAMESPACE -Dquickly
+          maven-command: clean install --errors --quiet -Pminikube -Dquarkus.container-image.tag=$GITHUB_SHA -Dnamespace=$NAMESPACE -Dquickly -Dquarkus.kubernetes.namespace=$NAMESPACE
```
Comment (danielezonca) [inline .github/workflows/Operator-CI.yaml:53]: Is there any reason why we have now the `$NAMESPACE` property in two different places? As far as I see `-Dnamespace` is a common property that works both in k8s and openshift

Comment (vaibhavjainwiz) [reply .github/workflows/Operator-CI.yaml]: `-Dquarkus.kubernetes.namespace` is common property k8s and openshift. I have removed `-Dnamespace`.

**Review by r00ta — COMMENTED** [review]
Not sure what you are trying to improve here. In addition to that it seems it does not work (see CI)

**Comment by vaibhavjainwiz** [comment]
CI is failing because 

> Not sure what you are trying to improve here. In addition to that it seems it does not work (see CI)

CI is failing because changes in `[Operator-CI.yaml](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/974/files#diff-9dd109889d5c5efdce83720ecb79d35f0ea6ed37f374e930fc6c7771e59fefd8)` file will not apply until it pushed.

**Comment by vaibhavjainwiz** [comment]
**Expectation**
For manage-tenant on-boarding, generated operator-bundle should be independent of namespaces. Namespace value will be injected while creating `addon`.

**Actual**
With the current implementation, namespace attribute always injected in to Deployment and ServiceAccount while compiling project using maven.

**Solution**
Remove hardcoded default namespace reference from pom.xml and only provide namespace value while compiling project using
`-Dquarkus.kubernetes.namespace=$NAMESPACE` argument.

File: shard-operator/pom.xml [hunk 2]
```diff
@@ -175,8 +175,6 @@
     <profile>
       <id>minikube</id>
       <properties>
-        <namespace>default</namespace>
-        <quarkus.kubernetes.namespace>${namespace}</quarkus.kubernetes.namespace>
```
Comment (robpblake) [inline shard-operator/pom.xml:179]: @vaibhavjainwiz How does this work for local-dev now? I don't see us setting this anywhere in the local environment or am I missing the configuration?

Comment (vaibhavjainwiz) [reply shard-operator/pom.xml:179]: when we run `shard-run.sh` or `mvn clean compile quarkus:dev` it uses the current namespace under `kubectl`.
```
<namespace>default</namespace>
<quarkus.kubernetes.namespace>${namespace}</quarkus.kubernetes.namespace>
```
is only for yaml generation - it sets the `namespace` in the resources generated under /target/generated-blablabla

**Review by sutaakar — APPROVED** [review]

**Comment by robpblake** [comment]
Recording my own explanation here:

When installing the Fleet Shard operator via the `AddOn` mechanism, there is a way to define the target namespace into which the resources in the `AddOn` should be created. If we hardcode the namespace into the resources, this will break the `AddOn` mechanism and therefore the namespace needs to be configurable.

**Review by robpblake — APPROVED** [review]

**Pull request merged**